aptos-move/move-examples/whitelist/sources/mint.move

```diff
@@ -0,0 +1,273 @@
+module whitelist_example::mint {
+    use std::error;
+    use std::signer;
+    use std::object;
+    use std::option;
+    use std::string::{Self, String};
+    use std::string_utils;
+    use aptos_framework::account::{Self, SignerCapability};
+    use aptos_framework::resource_account;
+    use aptos_token_objects::aptos_token::{Self, AptosToken};
+    use aptos_token_objects::collection::{Self, Collection};
+    use whitelist_example::whitelist::{Self};
+
+    struct MintConfiguration has key {
+        signer_capability: SignerCapability,
+    }
+
+    /// Action not authorized because the signer is not the owner of this module
+    const ENOT_AUTHORIZED: u64 = 1;
+
+    const COLLECTION_NAME: vector<u8> = b"Krazy Kangaroos";
+    const COLLECTION_DESCRIPTION: vector<u8> = b"A bunch of krazy kangaroos.";
+    const COLLECTION_URI: vector<u8> = b"https://krazykangaroos.io/collection-image-of-a-kangaroo";
+    const TOKEN_URI: vector<u8> = b"https://krazykangaroos.io/token-image-of-a-kangaroo";
+    const TOKEN_DESCRIPTION: vector<u8> = b"A single krazy kangaroo.";
+    const BASE_TOKEN_NAME: vector<u8> = b"Krazy Kangaroo";
+    const MAXIMUM_SUPPLY: u64 = 1000;
+    const MUTABLE_COLLECTION_DESCRIPTION: bool = false;
+    const MUTABLE_ROYALTY: bool = false;
+    const MUTABLE_URI: bool = false;
+    const MUTABLE_TOKEN_DESCRIPTION: bool = false;
+    const MUTABLE_TOKEN_NAME: bool = false;
+    const MUTABLE_TOKEN_PROPERTIES: bool = true;
+    const MUTABLE_TOKEN_URI: bool = false;
+    const TOKENS_BURNABLE_BY_CREATOR: bool = false;
+    const TOKENS_FREEZABLE_BY_CREATOR: bool = false;
+    const ROYALTY_NUMERATOR: u64 = 5;
+    const ROYALTY_DENOMINATOR: u64 = 100;
+    const U64_MAX: u64 = 18446744073709551615;
+    const WHITELIST_PRICE: u64 = 0;
+    const WHITELIST_START_TIME: u64 = 0;
+    const WHITELIST_END_TIME: u64 = 18446744073709551615;
+    const WHITELIST_PER_USER_LIMIT: u64 = 1;
+    const PUBLIC_PRICE: u64 = 1;
+    const PUBLIC_START_TIME: u64 = 0;
+    const PUBLIC_END_TIME: u64 = 18446744073709551615;
+    const PUBLIC_PER_USER_LIMIT: u64 = 2;
+
+    fun init_module(resource_signer: &signer) {
+        let resource_signer_cap = resource_account::retrieve_resource_account_cap(resource_signer, @owner);
+        move_to(resource_signer, MintConfiguration {
+            signer_capability: resource_signer_cap,
+        });
+    }
+
+    public entry fun initialize_collection(
+        owner: &signer,
+    ) acquires MintConfiguration {
+        assert!(signer::address_of(owner) == @owner, error::permission_denied(ENOT_AUTHORIZED));
+
+        let mint_configuration = borrow_global<MintConfiguration>(@whitelist_example);
+        let resource_signer = &account::create_signer_with_capability(&mint_configuration.signer_capability);
```

Comment:
How about we keep the whitelist as an object instead? The collection (not the focus of this example anyway) can be created with the owner being the creator

Reply:
Ah yes, that would make a lot more sense...too used to using resource accounts

Reply:
@movekevin you mean the publisher is also the creator of the contract and the owner of the whitelist object, yea? Just want to clarify

Reply:
The creator of the collection